### PR TITLE
Zephyr: fix build breakage with Zephyr "main"

### PR DESCRIPTION
**Pull request by lyakh**
Two new build regressions have been introduced in Zephyr's main, fix them.

### CODE REVIEW

**Review by pjdobrowolski — APPROVED** [review]

**Comment by kv2019i** [comment]
Thanks @lyakh! The timing is a bit unfortunate as we have this change before we managed to react to the previous one (so now if you want to update Zephyr, you'll need to touch multiple places in SOF). This needs to be merged together with a west update, but we can't do that before we have https://github.com/thesofproject/sof/pull/9190 merged.

**Comment by lyakh** [comment]
@kv2019i yes, let's wait until #9090 is merged and then I'll update this one with a manifest upgrade

**Comment by marc-hb** [comment]
Can you please add a short but actual commit message? At least the name of the functions changed. Or maybe even two commits to give enough room for something useful in commit titles.

File: src/include/sof/llext_manager.h
Comment (marc-hb) [inline src/include/sof/llext_manager.h:23]: Wow, this looks really weird.... I think it's not just "creative", unusual and harder to read...

- I don't think having these being either functions or macros depending on Kconfig is a good idea. Debugging Kconfig and preprocessing issues is painful enough already, so having totally different error messages depending on CONFIG_LLEXT seems wrong. Please keep these functions always: `static inline uintptr_tllext_manager_free_module(component_id) { return 0; }`. The generated code should be the same.

- Naive question sorry: why are these functions needed when CONFIG_LLEXT is false? I don't know all the details sorry but returning NULLs and zeros is an anti-pattern. Ideally, these shouldn't even be needed so a short comment/pointer explaining why would not hurt.

- where is `llext_unload()` defined when CONFIG_LLEXT is true?

Comment (lyakh) [reply src/include/sof/llext_manager.h:23]: > * Naive question sorry: why are these functions needed when CONFIG_LLEXT is false? I don't know all the details sorry but returning NULLs and zeros is an anti-pattern. Ideally, these shouldn't even be needed so a short comment/pointer explaining why would not hurt.

@marc-hb because they're called from lib_manager.c and that one is included in TGL builds too. No, I don't think it's needed there for the same reason as LLEXT - no memory mapping support.

> 
> * where is `llext_unload()` defined when CONFIG_LLEXT is true?

in Zephyr

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
ok, this one fixes the build on TGL which https://github.com/thesofproject/sof/pull/9217 is still pending. Lets go with this as it gets things back to green.

**Pull request merged**
